From 5cd792a71f02a995dec513c4066d37a817454205 Mon Sep 17 00:00:00 2001
From: Armin Novak <armin.novak@thincast.com>
Date: Wed, 25 Feb 2026 22:52:06 +0100
Subject: [PATCH 1/3] BACKPORT: UPSTREAM: [core,gateway] ignore incomplete rpc
 header

[Mingcong Bai: Fixed a minor post-3.23.0 merge conflict in
 libfreerdp/core/gateway/rpc_client.c]

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 libfreerdp/core/gateway/rpc_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libfreerdp/core/gateway/rpc_client.c b/libfreerdp/core/gateway/rpc_client.c
index 1f85820cc..93d874923 100644
--- a/libfreerdp/core/gateway/rpc_client.c
+++ b/libfreerdp/core/gateway/rpc_client.c
@@ -691,7 +691,7 @@ static SSIZE_T rpc_client_default_out_channel_recv(rdpRpc* rpc)
 			Stream_SetPosition(fragment, 0);
 
 			/* Ignore errors, the PDU might not be complete. */
-			rts_read_common_pdu_header(fragment, &header, TRUE);
+			(void)rts_read_common_pdu_header(fragment, &header, TRUE);
 			Stream_SetPosition(fragment, pos);
 
 			if (header.frag_length > rpc->max_recv_frag)
-- 
2.52.0

